File: level-3/answer.java
import java.util.*;
class temp{
    public static long solve(long m,long f,long m1,long f1,long count){
        if(m1==m && f1==f)
            return count;
        if(m1<m && f1==f)
            return solve(m,f,m1+f1,f1,count+1);
        if(m1==m && f1<f)
            return solve(m,f,m1,f1+m1,count+1);
        if(m1<m && f1<f)
            return Math.min(solve(m,f,m1+f1,f1,count+1),solve(m,f,m1,f1+m1,count+1));
        return Long.MAX_VALUE;
    }
    public static String solution(long m,long f){
        long count=solve(m, f, 1, 1, 0);
        if(count==Long.MAX_VALUE)
            return "impossible";
        return count+"";
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        long m=sc.nextLong();
        long f=sc.nextLong();
        System.out.println(solution(m,f));
        sc.close();
    }
}

/*The above solution is very slow and might not run for a few test cases*/

File: level-3/answer.py
def multiplier(m,f):
    diff=m-f
    return (diff//f)+1
def solution(x,y):
    m=int(x)
    f=int(y)
    count=0
    while True:
        if m<=0 or f<=0:
            break
        if m>100 or f>100:
            if m>f:
                mul=multiplier(m,f)
                m-=mul*f
                count+=mul
            elif f>m:
                mul=multiplier(f,m)
                f-=mul*m
                count+=mul
            else:
                break
        else:
            if m>f:
                m-=f
                count+=1
            elif f>m:
                f-=m
                count+=1
            else:
                break
    if m==1 and f==1:
        return str(count)
    return "impossible"

m=input()
f=input()
print(solution(m,f))
